docs/auth.md

```diff
@@ -0,0 +1,176 @@
+```
+
+### Local storage
+Local storage ([`window.localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage))
+is specific to a particular domain.
+
+```js
+// Get local storage and store as env variable
+const localStorage = await page.evaluate(() => JSON.stringify(window.localStorage));
+process.env.LOCAL_STORAGE = localStorage;
+
+// Set local storage in a new context
+const localStorage = process.env.LOCAL_STORAGE;
+await context.addInitScript(([storage]) => {
+  if (window.location.hostname === 'example.com') {
+    const entries = JSON.parse(storage);
+    Object.keys(entries).forEach(key => {
+      window.localStorage.setItem(key, entries[key]);
+    });
+  }
+}, [localStorage]);
+```
+
+### Session storage
+Session storage ([`window.sessionStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage))
+is specific to a particular domain.
+
+```js
+// Get session storage and store as env variable
+const sessionStorage = await page.evaluate(() => JSON.stringify(sessionStorage));
+process.env.SESSION_STORAGE = sessionStorage;
+
+// Set session storage in a new context
+const sessionStorage = process.env.SESSION_STORAGE;
+await context.addInitScript(([storage]) => {
+  if (window.location.hostname === 'example.com') {
+    const entries = JSON.parse(storage);
+    Object.keys(entries).forEach(key => {
+      window.sessionStorage.setItem(key, entries[key]);
+    });
+  }
+}, [sessionStorage]);
+```
+
+### Lifecycle
+
+1. Run tests (for example, with `npm run test`)
+2. Login via UI and retrieve authentication state
+    * In Jest, this can be executed in [`globalSetup`](https://jestjs.io/docs/en/configuration#globalsetup-string).
+3. In each test, load authentication state in `beforeEach` or `beforeAll` step.
+
+This lifecycle can be extended to cache authentication state across test runs.
+In that case, the expiry of authentication state (cookies/tokens in storage) would
+need to be handled.
+
+This approach to reuse authentication state would **work in CI environments**, since
+it does not rely on any external state.
+
+### Example
+
+[This script](examples/authentication.js) logs in on GitHub.com with Chromium,
+and then reuses the logged in cookie state in WebKit.
+
+### API reference
+- [class `BrowserContext`](./api.md#class-browsercontext)
+- [`browserContext.cookies`](./api.md#browsercontextcookiesurls)
+- [`browserContext.addCookies`](./api.md#browsercontextaddcookiescookies)
+- [`page.evaluate`](./api.md#pageevaluatepagefunction-arg)
+- [`browserContext.addInitScript`](./api.md#browsercontextaddinitscriptscript-arg)
+
+## Persistent authentication
+Web browsers use a directory on disk to store user history, cookies and other local
+state. This is directory is called the [User data directory](https://chromium.googlesource.com/chromium/src/+/master/docs/user_data_dir.md).
+
+User data directory is an alternate approach to reusing authentication state. It
```

Comment:
I'd rather advertise persistent context for persisting storage, e.g. IndexedDB. I think we should push towards non-persistent auth when we can.